### DISCUSSION TITLE
Exception occurred after updating to 2.18.0

### DISCUSSION THREAD

**Discussion by qian-o** [dotnet/Silk.NET#1744] [Q&A] [answered]
![image](https://github.com/dotnet/Silk.NET/assets/84434846/9786bd5e-e13a-4be3-ac75-1d226e5ca9f8)
![image](https://github.com/dotnet/Silk.NET/assets/84434846/d871508a-5205-422e-b056-02cf5782cd09)

How should I create imgui correctly and correctly ?

**Comment by Perksey** [comment]
Ensure the version of ImGui.NET you're using matches the version Silk.NET is using

**Reply by qian-o** [reply]
![image](https://github.com/dotnet/Silk.NET/assets/84434846/c891eae6-1e85-4c5f-a935-0dc6d2f1fdd1)
I did not install ImGui.NET separately, I still use Silk.NET.OpenGLES.Extensions.ImGui

**Reply by Perksey** [reply]
Try rebuilding your project (cleaning out bin/obj)

**Reply by qian-o** [reply]
![image](https://github.com/dotnet/Silk.NET/assets/84434846/d9c3e749-7c44-4574-9d12-3afc021751a4)
![image](https://github.com/dotnet/Silk.NET/assets/84434846/878ef19d-2098-4897-b14a-fe3d4c51bf43)
![image](https://github.com/dotnet/Silk.NET/assets/84434846/0cfcda1b-42c0-4b5f-832d-db25cb0f5f52)

I cleaned up all the content, but the error still exists.

**Reply by Perksey** [reply]
We’ve had another report of this, I have genuinely no idea what’s causing this. If someone can investigate this it’d be much appreciated.

**Reply by qian-o** [reply, marked as answer]
![image](https://github.com/dotnet/Silk.NET/assets/84434846/83e4b35f-7f8e-49c9-8327-ea1b635a64ab)
https://github.com/ImGuiNET/ImGui.NET/pull/412

I think the problem lies in ImGui.NET, but I'm not sure how to solve it.

**Reply by Perksey** [reply]
Yes, I've just vented here: <https://github.com/ImGuiNET/ImGui.NET/issues/436>
I recommend that you do the same to increase visibility.

**Reply by qian-o** [reply]
Okay, I will

**Reply by Beyley** [reply]
> I think the problem lies in ImGui.NET, but I'm not sure how to solve it.

A temporary solution you could do on your end, would be copying the ImGuiController code into your project, its a hack, but should make it work. A fix could be done on either side here, whether it be done by us through multi-targetting frameworks and providing a NS2.1 version of out `ImGuiController`s (so that NuGet grabs the correct one) and pushing out a hotfix, or pushing out a hotfix once ImGui.NET fixes this issue upstream
